data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/MultithreadedConsumer.java

```diff
@@ -74,14 +73,14 @@ public void run() {
 			MessageFormat schema = MessageFormat.getByMessageFormatByName(schemaType);
 			switch(schema){
 				case JSON:
-					KafkaSourceSchemaFactory.createConsumer(MessageFormat.JSON, jsonConsumer, status, buffer, topicConfig, sourceConfig, schemaType, pluginMetrics).consumeRecords();
+					new KafkaSourceCustomConsumer(jsonConsumer, status, buffer, topicConfig, sourceConfig, schemaType, pluginMetrics).consumeRecords();
```

Comment:
Not sure how this works. jsonConsumer, AvroConsumer and plaintextConsumer all three are initialized exactly same way -
```
this.jsonConsumer = new KafkaConsumer<>(consumerProperties);
		this.plainTextConsumer = new KafkaConsumer<>(consumerProperties);
		this.avroConsumer = new KafkaConsumer<>(consumerProperties);
```
How do they function differently? This is not the correct approach. You need to create them differently so that they can function differently

Reply:
We implemented the object initialization in an optimized way.